woocommerce/woocommerce-gutenberg-products-block
# Add filters for subtotalSuffix and saleBadgeSuffix

### Based against `fix/3735-filters-api`
- In `CartLineItemRow` execute the `subtotalPriceFormat` filter which extensions can use to to format the `ProductPrice` element that will be displayed under the product name. It must return a string containing `<price/>` so it can be used in `createInterpolateElement`.
- Modify `ProductPrice` so that it outputs the price in a way that follows the `format` it was passed.
- In `CartLineItemRow` execute the `saleBadgePriceFormat` filter which extensions can use to modify the _price_ of the `SaleBadge` component. It does not allow for the `Save` word to be changed currently.
- Add `format` prop to the `SaleBadge` component.
- Render the price in `SaleBadge` using `createInterpolateElement`
- Add tests for `ProductPrice` to test passing a format.
- 

Fixes #3747
Replaces #3763






### Screenshots
#### Before
![image](https://user-images.githubusercontent.com/5656702/106171210-34d94480-6189-11eb-8eda-3365a9d1da03.png)

#### After
![image](https://user-images.githubusercontent.com/5656702/106171336-52a6a980-6189-11eb-9475-33d70501f1bd.png)


#### Before 
![image](https://user-images.githubusercontent.com/5656702/107386565-e06c8800-6aeb-11eb-8d61-fa3b7b813e8b.png)

#### After
![image](https://user-images.githubusercontent.com/5656702/107386132-781da680-6aeb-11eb-981a-4e1794ac9f11.png)



### How to test the changes in this Pull Request:

1. Check out add/filter-subtotal-string-2 in blocks.
2. Add a non-subscription product to your cart with a sale price.
3. Add a subscription product to your cart with a sale price.
4. Add varying lengths/types of subscription product to your cart, every 2 weeks, every 6 months etc.
5. Open the cart block and verify the sale badge shows the period for subscriptions products, but not for regular products.
6. Verify the subtotal for subscription products shows under each product name and contains the period.
7. Verify the non-subscription product does not show a period for the subtotal.
